OrbitGl/PickingManager.cpp

```diff
@@ -7,8 +7,10 @@
 #include "OpenGl.h"
 #include "OrbitBase/Logging.h"
 
-PickingId PickingManager::CreateOrGetPickableId(
+PickingId PickingManager::GetOrCreatePickableId(
     std::weak_ptr<Pickable> pickable, BatcherId batcher_id) {
+  CHECK(!pickable.expired());
```

Comment:
auto pickable_ptr = pickable.lock();
CHECK(pickable_ptr);

On the other hand if we do not want this to be null - maybe just pass shared_ptr here instead of weak_ptr. CHECK with weak_ptr seems strange... Usually when weak_ptr used this case is handled.

Reply:
In this case we'd need to change more methods up to all the "Batcher::AddXXX" methods to accept a shared_ptr, and the reason for the weak_ptr there was to communicate that there won't be any ownership assumed. Nevertheless it does not make sense to pass an expired weak_ptr at this point... I'd prefer to leave the check, this should never happen but would be good to have just in case.

Reply:
ok, please then change the check so that weak pointer use is properly scoped.